src/stores/snapshot-slice.ts

```diff
@@ -355,10 +359,10 @@ function getAssociatedBulkOperation(
 	return [];
 }
 
-function isBulkOperationFulfilled( bulkOperation: BulkOperation ) {
+function isBulkOperationSettled( bulkOperation: BulkOperation ) {
 	return bulkOperation.operationIds.every( ( operationId ) => {
 		const operation = getOperation( operationId );
-		return operation?.status === 'fulfilled';
+		return operation && operation.status !== 'pending';
```

Comment:
This is not directly related to the issue I observed, but it adds more safety and could potentially fix similar problems in the future.